src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -52,78 +61,103 @@ public DrsService(SnapshotService snapshotService,
+                      ConfigurationService configurationService,
+                      KubeService kubeService,
                       PerformanceLogger performanceLogger) {
         this.snapshotService = snapshotService;
         this.fileService = fileService;
         this.drsIdService = drsIdService;
         this.samService = samService;
         this.resourceService = resourceService;
+        this.configurationService = configurationService;
+        this.kubeService = kubeService;
         this.performanceLogger = performanceLogger;
     }
 
-    public DRSObject lookupObjectByDrsId(AuthenticatedUserRequest authUser, String drsObjectId, Boolean expand) {
-
-        DrsId drsId = drsIdService.fromObjectId(drsObjectId);
-        SnapshotProject snapshotProject = null;
-        try {
-            UUID snapshotId = UUID.fromString(drsId.getSnapshotId());
-            // We only look up DRS ids for unlocked snapshots.
-            String retrieveTimer = performanceLogger.timerStart();
-
-            snapshotProject = snapshotService.retrieveAvailableSnapshotProject(snapshotId);
-
-            performanceLogger.timerEndAndLog(
-                retrieveTimer,
-                drsObjectId, // not a flight, so no job id
-                this.getClass().getName(),
-                "snapshotService.retrieveAvailable");
-        } catch (IllegalArgumentException ex) {
-            throw new InvalidDrsIdException("Invalid object id format '" + drsObjectId + "'", ex);
-        } catch (SnapshotNotFoundException ex) {
-            throw new DrsObjectNotFoundException("No snapshot found for DRS object id '" + drsObjectId + "'", ex);
+    private class DrsRequestResource implements AutoCloseable {
+        DrsRequestResource() {
+            // make sure not too many requests are being made at once
+            int podCount = kubeService.getActivePodCount();
+            int maxDRSLookups = configurationService.getParameterValue(ConfigEnum.DRS_LOOKUP_MAX);
+            int max = maxDRSLookups / podCount;
+            if (currentDRSRequests.incrementAndGet() > max) {
+                throw new TooManyRequestsException("Too many requests are being made at once. Please try again later.");
+            }
+            currentDRSRequests.incrementAndGet();
         }
 
-        // Make sure requester is a READER on the snapshot
-        String samTimer = performanceLogger.timerStart();
-
-        samService.verifyAuthorization(
-            authUser,
-            IamResourceType.DATASNAPSHOT,
-            drsId.getSnapshotId(),
-            IamAction.READ_DATA);
+        @Override
+        public void close() {
+            currentDRSRequests.decrementAndGet();
+        }
+    }
 
-        performanceLogger.timerEndAndLog(
-            samTimer,
-            drsObjectId, // not a flight, so no job id
-            this.getClass().getName(),
-            "samService.verifyAuthorization");
+    // throws TooManyRequestsException
```

Comment:
The data repo code base doesn't seem to have a lot of javadoc. It would be nice to add some as we go along. It doesn't have to be a lot of text. For this method, something like this would work: (I copied the text for `expand` from the open api yaml)
```suggestion
    /**
     * Look up the DRS object for a DRS object ID.
     *
     * @param authUser the user to authenticate this request for
     * @param drsObjectId the object ID to look up
     * @param expand if false and drsObjectId refers to a bundle, then the returned array contains only those
     *               objects directly contained in the bundle
     * @return the DRS object for this ID
     * @throws IamUnauthorizedException if the user is not authorized to look up this object
     * @throws TooManyRequestsException if there are too many concurrent DRS lookup requests
     */
```